### PR TITLE
Allow to skip NVM installation in setup-local-env.sh

### PR DESCRIPTION
**Pull request by desaiuditd**
## Description
Related to #9120.

I'm using [nodenv](https://github.com/nodenv/nodenv/) as Node Version Manager. So it's conflicting with the NVM setup script in `./setup-local-env.sh`.

This PR introduces a flag option for the `setup-local-env.sh` script to skip NVM installation and setup, if needed. By default, it will behave as is.

## How has this been tested?

- Running `./bin/setup-local-env.sh --skip-nvm-setup` on the local machine will skip all the steps related to NVM, and directly install `npm` packages.

- So far, I've only tested this on my Mac environment. I've used [`getopts`](https://ss64.com/bash/getopts.html) which seems to be in-built command in bash. So hopefully, it also works in Windows/Linux. Happy to listen from other OS users.

- I've also updated the Getting Started doc for dev contributors.

## Types of changes

- Local Environment Setup script.
- Documentation to let people know about the new flag to skip NVM installation.

## Checklist:
- [x] My code is tested (Only on MacOS so far. Need help in testing this on other OS).
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by desaiuditd** [comment]
Reviewers' Note: Hide whitespace changes while reviewing. It will help to understand the new change in better way.

File: bin/bootstrap-env.sh [hunk 1]
```diff
@@ -10,3 +10,16 @@ DOCKER_COMPOSE_FILE_OPTIONS="-f docker-compose.yml"
 if [ "$DOCKER_ENV" == "localwpdev" ]; then
 	DOCKER_COMPOSE_FILE_OPTIONS="${DOCKER_COMPOSE_FILE_OPTIONS} -f docker-compose-localdev.yml"
 fi
+
+# Check for skip-nvm-setup flag. Used in install-node-nvm.sh
+while getopts ":-:" opt; do
+	case "${opt}" in
+		-)
+			case "${OPTARG}" in
+				skip-nvm-setup)
```
Comment (desaiuditd) [inline bin/bootstrap-env.sh]: Open to suggestions on renaming this flag name.

**Comment by gziolo** [comment]
This change breaks Travis jobs causing them to run forever, see:
https://travis-ci.com/WordPress/gutenberg/builds/116557184

Screen Shot 2019-06-24 at 09 39 44

**Comment by desaiuditd** [comment]
First commit was indeed a genuine mistake from my side and the build was failing because of that.

But last 2 builds doesn't look like relevant failures. In first, an e2e test failed  and in second, a unit test failed.

I am not able to conclude a straight forward reason, why they would fail due to my changes, which are essentially targeted towards build tooling and local development.

Can someone have a second look? 👀

**Comment by desaiuditd** [comment]
Merged latest master into my branch and now tests are passing without any changes from my end. Should be good to review now.

**Comment by senadir** [comment]
cc @pento does this sound like something we might want to have?
I have concerns about possible problems if we don't have a way to check for the node version used

**Comment by youknowriad** [comment]
nvm is not really a standard to check or install node versions, I'd personally just remove all nvm related scripts... and just check if the node version is not right, ask the person to install the right one with its preferred method.

**Comment by pento** [comment]
Yah, NVM was an interesting experiment in trying to ensure the correct version of Node was installed, but I don't think it was a successful as I would've liked.

#17004 will remove all of these scripts, so I'm going to close this PR in favour of that one.

**Pull request closed without merging**